smart-contracts/contracts/interfaces/IPublicLock.sol

```diff
@@ -269,21 +269,24 @@ contract IPublicLock is IERC721Enumerable {
     uint amount
   ) external;
 
-  /**
-   * @notice Destroys the msg.sender's key and sends a refund based on the amount of time remaining.
+   /**
+   * @dev Destroys the key managed by msg.sender and sends a refund
```

Comment:
instead of `by msg.sender` it's the token id provided right?

Reply:
@NickCuso So it's both. As per the permissions doc, `cancelAndRefund` is only callable by the keyManager of a given key. This is in line with what we had before, where a key owner could only cancel their own key. Now, someone can only cancel a key they manage (unless using metaTX and  `cancelAndRefundFor`
We need the token ID here because we have no way to look up the token given only the keyManager.

Reply:
Ah, but I could still change the comment to make it simpler. The implementation is as above though.

Reply:
yup makes sense. but for clarity it's based on the tokenId since the keyManager may have control over may different keys

Reply:
Maybe something like "allows the key manager to expire a given tokenId and send a refund to the keyOwner"

Reply:
Exactly. Fixed.